src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,352 @@
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  borderRadius({ theme }) {
+    return theme.spacings.xxsmall;
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${styles.borderRadius};
+  border: ${styles.border};
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+  testID,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const buttonColor = `${variantColor || 'primary'}`;
+
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw Error(
+      `Error in Button: expected \`children\` of type \`string\` but found ${typeof children}`,
+    );
+  }
+
+  return (
+    <Flex
+      flex={block ? 1 : 0}
+      flexDirection="row"
+      alignItems="center"
+      alignSelf={styles.align({ align })}
+      justifyContent="center"
+    >
+      <Space padding={styles.padding({ size, children, theme })}>
+        <StyledButton
+          buttonColor={buttonColor}
+          onPress={onClick}
+          variantColor={variantColor}
+          disabled={disabled}
+          size={size}
+          block={block}
+          variant={variant}
+          activeOpacity={1}
+          underlayColor={styles.underlayColor({ variant, buttonColor, theme })}
+          {...automation(testID)}
+        >
+          <React.Fragment>
+            {icon && iconAlign === 'left' && (
+              <Icon
+                name={icon}
+                size={styles.iconSize({ size, children })}
+                fill={styles.fontColor({ variant, variantColor, disabled })}
+              />
+            )}
+            {icon && iconAlign === 'left' && children && (
+              <Space margin={styles.spaceBetween({ size, iconAlign })}>
+                <View />
+              </Space>
+            )}
+            {children && (
+              <Text
+                color={styles.fontColor({ variant, variantColor, disabled })}
+                size={styles.fontSize({ size })}
+                align="center"
+                _weight="bold"
+                _letterSpacing={styles.letterSpacing({ size })}
+                _lineHeight={styles.lineHeight({ size })}
+              >
+                {size === 'xsmall' ? children.toUpperCase() : children}
```

Comment:
what if children is not present? the `upperCase` will throw an error. Shouldn't this be `size === 'xsmall' && children ? children.toUpperCase() : children`

Reply:
I have added the children check inline 296 and removed it from here. It looks cleaner now.

Reply:
`condition ? render component: null`